core/drivers/clk/sam/clk-sam9x60-pll.c

```diff
@@ -0,0 +1,502 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ *  Copyright (C) 2019 Microchip Technology Inc.
+ *
+ */
+
+#include <io.h>
+#include <kernel/delay.h>
+#include <mm/core_memprot.h>
+#include <types_ext.h>
+#include <util.h>
+#include "at91_clk.h"
+
+#define PMC_PLL_CTRL0_DIV_MSK 0xf
+#define PMC_PLL_CTRL0_DIV_POS 0
+#define	PMC_PLL_CTRL1_MUL_MSK 0xff
+#define	PMC_PLL_CTRL1_MUL_POS 24
+#define	PMC_PLL_CTRL1_FRACR_MSK	0x3fffff
+#define	PMC_PLL_CTRL1_FRACR_POS	0
+
+#define PLL_STATUS_MASK(id)	BIT(1 + (id))
+#define PLL_REG(id)		(AT91_CKGR_PLLAR + ((id) * 4))
+#define PLL_DIV_MASK		0xff
+#define PLL_DIV_MAX		PLL_DIV_MASK
+#define PLL_DIV(reg)		((reg) & PLL_DIV_MASK)
+#define PLL_MUL(reg, layout) ({ \
+	typeof(layout) __layout = layout; \
```

Comment:
add an empty line between variable definition and instructions:
```patch
 #define PLL_MUL(reg, layout) ({ \
 	typeof(layout) __layout = layout; \
+	\
 	(((reg) >> (__layout)->mul_shift) & (__layout)->mul_mask); \
```

Also: prefer s/reg/_reg/g and s/layout/_layout/ to prevent collision with variables `reg` or `layout` defined in the scope of the function that uses the macro.